main_window/rocket_data.py

```diff
@@ -157,19 +157,22 @@ def last_value_and_time(self, device: DeviceType, data_entry_id: DataEntryIds) -
         :return: Value, Time
         :rtype:
         """
-        ret = self.time_series_by_device(device, data_entry_id)
-
-        if ret is None:
-            return None
+        with self.data_lock:
+            all_times = list(self.timeset.keys())
```

Comment:
`.keys()` supports `reversed()`, so no need to convert into list. Same applies for other list(keys()) call.